clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -420,7 +420,7 @@ private static int parseAcks(String acksString) {
      *
      * @throws InterruptException If the thread is interrupted while blocked
      * @throws SerializationException If the key or value are not valid objects given the configured serializers
-     * @throws BufferExhaustedException If <code>block.on.buffer.full=false</code> and the buffer is full.
+     * @throws TimeoutException if the time taken for fetching metadata, serialization of (key,value), partitioning has surpassed <code>max.block.ms</code>.
```

Comment:
oh ok, In that case I am planning to change the doc to :   
@throws TimeoutException if the time taken for fetching metadata, allocating memory for the record has surpassed max.block.ms.